src/paths.rs

```diff
@@ -81,8 +81,18 @@ impl<P: PointScaler> Paths<P> {
     }
 
     /// Construct a scaled clone of the path with the origin at the path center.
-    pub fn scale(&self, scale: f64) -> Self {
-        Self::new(self.0.iter().map(|p| p.scale(scale)).collect())
+    pub fn scale(&self, scale_x: f64, scale_y: f64) -> Self {
```

Comment:
I like the general direction to add both the ability to choose the center point to scale around. And also why now allow a different value for the x and y scaling.

The thoughts I have is that it might be more common that you want to scale with x and y as the same value that varying these. We could solve this with keeping the signature .scale(&self, scale: f64), and add .scale_xy(&self, scale_x: f64, scale_y: f64). But that then also might require us to add  scale_xy_around_point as well and then we have 4 different scaling methods which is quite a lot.

We could also maybe skip one and go for the three variations `scale(&self, scale: f64)`, `scale_xy(&self, scale_x: f64, scale_y: f64)`, and `scale_around_point(&self, scale_x: f64, scale_y: f64, point: Point<P>)`

Any thoughts around this?

Reply:
Oh, and we should probably have like one test around each of these methods to ensure they work as intended.

Reply:
> And also why now allow a different value for the x and y scaling.

I am using non-uniform scaling when transforming polygons between Y-up and Y-down coordinate spaces. I figure that since the corresponding scale methods in Clipper2 take two arguments (e.g. https://github.com/tirithen/clipper2c-sys/blob/b8478fec9d81857e158f8a242fd441d01bad844f/clipper2c/include/clipper2c.h#L154) this would be a reasonable API here too.

> We could also maybe skip one and go for the three variations scale(&self, scale: f64), scale_xy(&self, scale_x: f64, scale_y: f64), and scale_around_point(&self, scale_x: f64, scale_y: f64, point: Point<P>)

I'm fine with this approach!

> Oh, and we should probably have like one test around each of these methods to ensure they work as intended.

I can add tests for this

Reply:
@msvbg thanks for pointing out that the Clipper2 API uses two arguments, I agree that we should stay as close as possible to that API for recognition with that library so lets go with your signatures `scale(&self, scale_x: f64, scale_y: f64)`, and `scale_around_point(&self, scale_x: f64, scale_y: f64, point: Point<P>)`.

Its a breaking change so the commit message should be annotated [`feat!: add scale...` and a `BREAKING CHANGE:` footer](https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-both--and-breaking-change-footer). For now I'm using the `standard-version` tool to generate the CHANGELOG.md from the commit messages, I need to read up more on the Rust tooling for this soonish, but they should work with the same commit formats I think.